napari/layers/vectors/vectors.py

```diff
@@ -28,6 +32,9 @@ class Vectors(Layer):
         D dimensions. An (N1, N2, ..., ND, D) array is interpreted as
         "image-like" data where there is a length D vector of the
         projections at each pixel.
+    ndim : int
+        Number of dimensions for vectors. When data is not None, ndim must be D.
```

Comment:
```suggestion
        Number of dimensions for vectors. When data is not None, ndim defaults to D.
```

Reply:
I copied from the Points layer here... If the required `D` differs from the data, it won't change the data but simply raise an error.

Reply:
okay cool, consistency is most important I guess!

Reply:
Though I agree that this might be a good idea... follow up PR maybe :)

Reply:
The more I think about this, the less I like it. I don't think there are any safe assumption to what a user would expect to happen when passing lowe dimensionality data. Prepend zeros if ndim is >3, but append them if it's lower (due to xyz ordering)? And there's probably a lot of cases where one would *want* the failure.

Reply:
If we were to do it I would imagine following numpy broadcasting rules and prepending would be the way to go, but agree that erroring is the safest solution for now!

Reply:
> If we were to do it I would imagine following numpy broadcasting rules and prepending would be the way to go, but agree that erroring is the safest solution for now!

:100: 

@brisvag we don't do xyz. ;)

Reply:
Yeah, I agree, my point was that people *might* expect that, but get something different... In which case, I prefer the error :P